app/assets/javascripts/scan_analytics.js

```diff
@@ -0,0 +1,210 @@
+$(document).ready(function(){
+  scanDataFetch()
+});
+function scanDataFetch(){
+  $.ajax({
+    type: 'GET',
+    url: '/p/' + $('#chart_data').data('project-id') + '/scan_analytics/charts?scan_id=' + $('#chart_data').data('scan-project-id'),
+    success: function(data) {
+      var options = {
+        chart: {
+          zoomType: 'xy'
+        },
+      };
+      Highcharts.setOptions({
```

Comment:
Move out logic into independent functions. Right now we are doing a bunch of different operations in the same block. It would be better to segregate it into well named functions.